app/models/watched_word.rb

```diff
@@ -40,6 +40,9 @@ def self.actions
   after_destroy :clear_cache
 
   scope :by_action, -> { order("action ASC, word ASC") }
+  scope :for, ->(word:) do
+    where("(word ILIKE :word AND case_sensitive = 'f') OR (word LIKE :word AND case_sensitive = 't')", word: word)
```

Comment:
```suggestion
    where("(NOT case_sensitive AND word ILIKE :word) OR (case_sensitive AND word LIKE :word)", word: word)
```

It'll be more efficient if we check the boolean column first such that every row will only either do a `LIKE` or `ILIKE` filter.

Reply:
Do you have any proofs of that? Have you tested it? Isn't the query planner not clever enough to do this optimisation on the fly?

Reply:
Ah I’m wrong here, Postgres doesn’t have short circuiting in the where clause and it is up to the planner to decide. https://stackoverflow.com/a/14944611

Sorry ignore my comment here please.

Reply:
Ahah, I guess it is clever enough to not take this into account 😅

Reply:
We can influence with an index though but I don’t think it’ll be worth it on this table which likely has very few rows.